src/fn_call.rs

```diff
@@ -1203,6 +1206,27 @@ pub fn run_builtin_binary_op(
         }
     }
 
+    #[cfg(feature = "decimal")]
```

Comment:
I don't know if a `Decimal` can overflow or underflow.  You may want to actually copy the `INT` code block instead of the `FLOAT` code block if you need checked arithmetic...  At least the division should trap divide-by-zero...

Reply:
I'll have to take a look at how this all fits together a bit closer. `Decimal` can overflow, but it has checked versions for add, sub, etc that return an Option. 

Rhai also has Checked traits that return an Option, so all good there. I'll see later on today if I can use those traits for the `Decimal` type.

Reply:
Yes, look into the `add`, `sub` etc. functions.  Those are checked versions.  You can copy that template for your own `add_decimal` etc.  Essentially if it overflows, convert it into an error.

Reply:
Well, actually I see that `Decimal` seems to implement `num_traits::CheckedAdd` etc.  If so, maybe you can simply use the generic versions of `add`, `sub` etc. and no need to write your own versions.